packages/pegasus/src/once-promise-kit.js

```diff
@@ -0,0 +1,33 @@
+import { assert } from '@agoric/assert';
+import { makePromiseKit } from '@agoric/promise-kit';
+
+/**
+ * Create a promise kit that will throw an exception if it is resolved or
+ * rejected more than once.
+ *
+ * @param {() => Details} makeReinitDetails
+ */
+export const makeOncePromiseKit = makeReinitDetails => {
+  const { promise, resolve, reject } = makePromiseKit();
+
+  let initialized = false;
```

Comment:
```suggestion
  let resolved = false;
```
Calling either `resolve` or `reject` resolves the promise.